django/middleware/security.py

```diff
@@ -20,10 +20,9 @@ def __init__(self, get_response):
 
     def process_request(self, request):
         path = request.path.lstrip("/")
-        if (
-            self.redirect
-            and not request.is_secure()
-            and not any(pattern.search(path) for pattern in self.redirect_exempt)
+        if self.redirect and not (
+            request.is_secure()
+            and any(pattern.search(path) for pattern in self.redirect_exempt)
```

Comment:
(!x && !y) is not the same thing as !(x && y)

Please see De Morgan's laws: https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
Thanks, i didn't pay attention to it, you are right